sdk/storage/inc/datalake/path_client.hpp

```diff
@@ -0,0 +1,249 @@
+
+    /**
+     * @brief Sets the owner, group, permissions, or access control list for a file or directory.
+     *        Note that Hierarchical Namespace must be enabled for the account in order to use
+     *        access control. Also note that the Access Control List (ACL) includes permissions for
+     *        the owner, owning group, and others, so the x-ms-permissions and x-ms-acl request
+     *        headers are mutually exclusive.
+     * @param options Optional parameters to set an access control to the resource the path points
+     *                to.
+     * @return PathFlushDataResponse
+     */
+    PathSetAccessControlResponse SetAccessControl(
+        const SetAccessControlOptions& options = SetAccessControlOptions()) const;
+
+    /**
+     * @brief Sets POSIX access control rights on files and directories under given path
+     *        recursively.
+     * @param mode Mode PathSetAccessControlRecursiveMode::Set sets POSIX access control rights on
+     *             files and directories, PathSetAccessControlRecursiveMode::Modify modifies one or
+     *             more POSIX access control rights  that pre-exist on files and directories,
+     *             PathSetAccessControlRecursiveMode::Remove removes one or more POSIX access
+     *             control rights that were present earlier on files and directories
+     * @param options Optional parameters to set an access control recursively to the resource the
+     *                path points to.
+     * @return PathSetAccessControlRecursiveResponse
+     */
+    PathSetAccessControlRecursiveResponse SetAccessControlRecursive(
+        PathSetAccessControlRecursiveMode mode,
+        const SetAccessControlRecursiveOptions& options = SetAccessControlRecursiveOptions()) const;
+
+    /**
+     * @brief Sets the properties of a resource the path points to.
+     * @param options Optional parameters to set the properties to the resource the path points to.
+     * @return PathUpdateResponse
+     */
+    PathUpdateResponse SetProperties(
+        const SetPathPropertiesOptions& options = SetPathPropertiesOptions()) const;
+
+    /**
+     * @brief Creates a file or directory. By default, the destination is overwritten and
+     *        if the destination already exists and has a lease the lease is broken.
+     * @param options Optional parameters to create the resource the path points to.
+     * @return PathCreateResponse
+     */
+    PathCreateResponse Create(const PathCreateOptions& options = PathCreateOptions()) const;
+
+    /**
+     * @brief Renames a file or directory. By default, the destination is overwritten and
+     *        if the destination already exists and has a lease the lease is broken.
+     * @param options Optional parameters to rename a resource to the resource the path points to.
+     * @return PathRenameResponse
+     */
+    PathRenameResponse Rename(const PathRenameOptions& options = PathRenameOptions()) const;
```

Comment:
it did strike me that `Rename` function doesn't have a mandatory parameter. It makes me feel that i'm calling `move` on Linux with only one argument.